packages/core/core-flows/src/common/workflows/batch-links.ts

```diff
@@ -0,0 +1,32 @@
+import { LinkDefinition } from "@medusajs/modules-sdk"
+import { BatchWorkflowInput } from "@medusajs/types"
+import {
+  WorkflowData,
+  createWorkflow,
+  parallelize,
+} from "@medusajs/workflows-sdk"
+import { createRemoteLinkStep } from "../steps/create-remote-links"
+import { dismissRemoteLinkStep } from "../steps/dismiss-remote-links"
+import { updateRemoteLinksStep } from "../steps/update-remote-links"
+
+export const batchLinksWorkflowId = "batch-links"
+export const batchLinksWorkflow = createWorkflow(
+  batchLinksWorkflowId,
+  (
+    input: WorkflowData<
+      BatchWorkflowInput<LinkDefinition, LinkDefinition, LinkDefinition>
+    >
+  ) => {
+    const [created, updated, deleted] = parallelize(
+      createRemoteLinkStep(input.create || []),
+      updateRemoteLinksStep(input.update || []),
+      dismissRemoteLinkStep(input.delete || [])
```

Comment:
suggestion: Can we check in all the steps that there is an input and otherwise return prematurely?

Reply:
all of them are :)